conformance/lang/expressions/error_constructor.balt

```diff
@@ -0,0 +1,526 @@
+    io:println(err2.cause() == ()); // @output false
+    io:println(err2.cause() === err); // @output true
+
+    error err3 = error E("error message 3");
+    io:println(err3.message() == "error message 3"); // @output true
+    io:println(err3.cause() == ()); // @output true
+
+    error err4 = error E("error message 4", err);
+    io:println(err4.message() == "error message 4"); // @output true
+    io:println(err4.cause() == ()); // @output false
+    io:println(err4.cause() === err); // @output true
+}
+
+Test-Case: output
+Description: Test evaluating the error-constructor-expr constructs a new immutable detail mapping,
+             where each named-arg specifies a field of the error detail mapping and
+             the static type of each named-arg must be a subtype of value:Cloneable.
+Labels: error-constructor-expr, module-type-defn, record-type, error-type, string, is-expr, map-type, intersection-type,
+         readonly, error:detail, value:Cloneable, method-call-expr, list-constructor-expr, DecimalFloatingPointNumber
+
+type E error<record {|string message; boolean fatal?;|}>;
+
+function init() {
+    error err1 = error("error message 1");
+    io:println(err1.detail() is map<value:Cloneable>); // @output true
+    io:println(err1.detail() is readonly); // @output true
+    io:println(err1.detail()); // @output {}
+
+    error err = error("ERROR!");
+    error err2 = error("error message 2", err);
+    io:println(err2.detail() is map<value:Cloneable>); // @output true
+    io:println(err2.detail() is readonly); // @output true
+    io:println(err2.detail()); // @output {}
+
+    map<string> m = {"x": "x", "y": "y"};
+    error err3 = error("error message 3", a = "a", b = false, c = 1, d = 1.3, e = err, f = [1, 2, 3], g = m);
+    io:println(err3.detail() is map<value:Cloneable>); // @output true
+    io:println(err3.detail() is readonly); // @output true
+    io:println(err3.detail()); // @output {"a":"a","b":false,"c":1,"d":1.3,"e":error("ERROR!"),"f":[1,2,3],"g":{"x":"x","y":"y"}}
+
+    error err4 = error("error message 4", err, a = "a", b = false, c = 1, d = 1.3, e = err, f = [1, 2, 3], g = m);
+    io:println(err4.detail() is map<value:Cloneable>); // @output true
+    io:println(err4.detail() is readonly); // @output true
+    io:println(err4.detail()); // @output {"a":"a","b":false,"c":1,"d":1.3,"e":error("ERROR!"),"f":[1,2,3],"g":{"x":"x","y":"y"}}
+
+    error err5 = error E("error message 1", message = "Oops!");
+    io:println(err5.detail() is map<value:Cloneable>); // @output true
+    io:println(err5.detail() is readonly); // @output true
+    io:println(err5.detail()); // @output {"message":"Oops!"}
+
+    error err6 = error E("error message 2", err, message = "Oops!", fatal = true);
+    io:println(err6.detail() is readonly); // @output true
+    io:println(err6.detail() is map<value:Cloneable>); // @output true
+    io:println(err6.detail()); // @output {"message":"Oops!","fatal":true}
```

Comment:
Shall we also do type test for the named args to check whether they are of a subtype of `value:Cloneable`?
```ballerina
    io:println(err6.detail()["message"] is readonly); 
    io:println(err6.detail()["message"] is value:Cloneable);
```